### PR TITLE
feat: manifest command group for adding and listing flags in manifest

### LINKED ISSUE #153: feat(cli): manifest command with add flag
# feat(cli): manifest command with add flag

Add a new `manifest` command to the CLI that allows users to manage their flag manifest file directly from the command line. This will provide a more streamlined workflow for flag management without requiring manual file editing.

Starting with the `add` sub-command to add new flags to the manifest file, with plans to expand to other manifest management operations in the future.

## Acceptance Criteria

- [ ] Implement `openfeature manifest add` command
- [ ] Command requires flag name and default value as mandatory parameters
- [ ] Command defaults type to "boolean" but allows override via optional type parameter
- [ ] Command accepts optional description parameter
- [ ] Command validates flag configuration before adding to manifest
- [ ] Use temp file pattern for safe I/O operations:
  - [ ] Write updated manifest to temporary file
  - [ ] Complete all I/O operations on temp file
  - [ ] Only overwrite original manifest if no errors occur
- [ ] Command provides clear success/error feedback
- [ ] Command follows existing CLI patterns and conventions
- [ ] Add appropriate help text and usage examples showing required vs optional parameters
- [ ] Include unit tests for the new functionality

## Notes

- Required parameters: name, default value
- Optional parameters: type (defaults to "boolean"), description
- Follow existing temp file pattern used elsewhere in codebase for safe file operations
- This is the first step in a broader manifest management feature set
- Future sub-commands may include: `remove`, `list`, `update`, `validate`
- Should integrate with existing manifest file format and validation logic

### PR DESCRIPTION
**Pull request by georgesdugue**
## This PR

  - Adds `openfeature manifest` command group for managing flag manifest files
  - Adds `manifest add` subcommand to add new flags to manifests with type validation and duplicate detection
  - Adds `manifest list` subcommand to display all flags in a formatted table view
  - Implements comprehensive type validation for all five flag types (boolean, string, integer, float, object)
  - Automatically creates new manifest files if they don't exist
  - Displays updated flag list after successful additions

  ### Related Issues

  Fixes #153

  ### Notes

  - The `manifest add` command validates default values against the specified flag type before adding to the manifest
  - Long descriptions and values are automatically truncated in the list view for better readability
  - Both commands include extensive unit test coverage with edge case handling

  ### Follow-up Tasks

  None

  ### How to test

  **Testing `manifest add`:**
  ```bash
  # Add a boolean flag
  openfeature manifest add new-feature --default-value false --description "Enable new feature"

  # Add a string flag
  openfeature manifest add welcome-message --type string --default-value "Hello!" --description "Welcome message for users"

  # Add an integer flag
  openfeature manifest add max-retries --type integer --default-value 3

  # Add a float flag
  openfeature manifest add discount-rate --type float --default-value 0.15

  # Add an object flag
  openfeature manifest add config --type object --default-value '{"key":"value"}'

  # Test error handling - try adding duplicate flag
  openfeature manifest add new-feature --default-value true
  # Should error: "flag 'new-feature' already exists in the manifest"

### CODE REVIEW

File: internal/cmd/manifest_add_test.go [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+				"add", "discount-rate",
+				"--type", "float",
+				"--default-value", "0.15",
+			},
+			existingManifest: `{
+				"$schema": "https://raw.githubusercontent.com/open-feature/cli/refs/heads/main/schema/v0/flag-manifest.json",
+				"flags": {}
+			}`,
+			validateResult: func(t *testing.T, fs afero.Fs) {
+				content, err := afero.ReadFile(fs, "flags.json")
```
Comment (kriscoleman) [inline internal/cmd/manifest_add_test.go:115]: I'm loving all of these tests! Very robust testing! 

This doesn't have to change, but just some food for thought: 
a lot of these values like flags.json, existingManifest, flags: {}, etc. are all repeated in these tests and constant. They may be able to be pulled up into the t.Run below that executes `validateResult`, and that might reduce the lines of code a bit, but it would trade off the ability to customize those values for different test cases if we ever wanted to.

**Comment by georgesdugue** [comment]
> approved!
> 
> we'll just need to address the pr lint and pr test failures @georgesdugue

Thank you! Will address the failing checks soon.
